internal/wasmldr/src/loader/mod.rs

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: Apache-2.0
+
+mod acquired;
+mod attested;
+mod compiled;
+mod configured;
+mod connected;
+mod pki;
+mod requested;
+
+use std::sync::Arc;
+
+use rustls::{ClientConfig, ServerConfig};
+use wasi_common::WasiCtx;
+use wasmtime::{Linker, Store, Val};
+use zeroize::Zeroizing;
+
+use crate::config::Config;
+
+pub struct Configured {
+    config: Config,
+}
+
+pub struct Requested {
+    config: Config,
+    prvkey: Zeroizing<Vec<u8>>,
+    crtreq: Vec<u8>,
+}
+
+pub struct Attested {
+    config: Config,
+    srvcfg: Arc<ServerConfig>,
+    cltcfg: Arc<ClientConfig>,
+}
+
+pub struct Acquired {
+    config: Config,
+    srvcfg: Arc<ServerConfig>,
+    cltcfg: Arc<ClientConfig>,
+
+    webasm: Vec<u8>,
+}
+
+pub struct Compiled {
+    config: Config,
+    srvcfg: Arc<ServerConfig>,
+    cltcfg: Arc<ClientConfig>,
+
+    wstore: Store<WasiCtx>,
+    linker: Linker<WasiCtx>,
+}
+
+pub struct Connected {
+    wstore: Store<WasiCtx>,
+    linker: Linker<WasiCtx>,
+}
+
+pub struct Completed {
+    values: Vec<Val>,
+}
```

Comment:
How would you feel about defining the above structs in their respective files?

Reply:
Defining them in their respective files means that I need to make all the fields `pub(super)`. Both have drawbacks.